hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -163,14 +200,93 @@ public AbfsOutputStream(AbfsOutputStreamContext abfsOutputStreamContext)
         abfsOutputStreamContext.getSasTokenRenewPeriodForStreamsInSeconds());
     this.outputStreamId = createOutputStreamId();
     this.tracingContext = new TracingContext(abfsOutputStreamContext.getTracingContext());
+    if (prefixMode == PrefixMode.BLOB) {
+      // Get the list of all the committed blocks for the given path.
+      this.committedBlockEntries = getBlockList(path, tracingContext);
+    }
     this.tracingContext.setStreamID(outputStreamId);
     this.tracingContext.setOperation(FSOperationType.WRITE);
     this.ioStatistics = outputStreamStatistics.getIOStatistics();
     this.blockFactory = abfsOutputStreamContext.getBlockFactory();
     this.blockSize = bufferSize;
+    this.prefixMode = client.getAbfsConfiguration().getPrefixMode();
```

Comment:
line 203 is already using prefix mode, while the local variable is being set in this line. Ordering needs fixing ?

Reply:
Fixed.